programs/validator-history/src/utils.rs

```diff
@@ -1,13 +1,21 @@
 use anchor_lang::{
-    prelude::{AccountInfo, Pubkey},
+    prelude::{AccountInfo, Pubkey, Result},
+    require,
     solana_program::native_token::lamports_to_sol,
 };
 
-pub fn cast_epoch(epoch: u64) -> u16 {
-    (epoch % u16::MAX as u64).try_into().unwrap()
+use crate::errors::ValidatorHistoryError;
+
+pub fn cast_epoch(epoch: u64) -> Result<u16> {
+    require!(
+        epoch < (u16::MAX as u64),
+        ValidatorHistoryError::EpochTooLarge
+    );
+    let epoch_u16: u16 = (epoch % u16::MAX as u64).try_into().unwrap();
```

Comment:
nit: can't you do try_into()? unwrap in fn with Result<>

Reply:
the Result here is secretly Result<T, AnchorError> which is incompatible with the try_into return type here but since we know it's gonna be a valid u16 from the line above the unwrap is fine